sed/loader/flash/loader.py

```diff
@@ -922,7 +255,45 @@ def read_dataframe(
                 metadata=metadata,
             )
 
-        df, df_timed = self.parquet_handler(data_parquet_dir, **kwds)
+        # if parquet_dir is None, use data_parquet_dir
+        parquet_dir = parquet_dir or data_parquet_dir
+        parquet_path = Path(parquet_dir)
+        filename = "_".join(str(run) for run in self.runs)
+        converted_str = "converted" if converted else ""
+        # Create parquet paths for saving and loading the parquet files of df and timed_df
+        ph = ParquetHandler(
+            [filename, filename + "_timed"],
+            parquet_path,
+            converted_str,
+            "run_",
+            detector,
+        )
+
+        # Check if load_parquet is flagged and then load the file if it exists
+        if load_parquet:
+            df_list = ph.read_parquet()
+            df = df_list[0]
+            df_timed = df_list[1]
+
+        # Default behavior is to create the buffer files and load them
+        else:
+            # Obtain the parquet filenames, metadata, and schema from the method
+            # which handles buffer file creation/reading
+            h5_paths = [Path(file) for file in self.files]
+            buffer = BufferHandler(
+                self._config["dataframe"],
+                h5_paths,
+                parquet_path,
+                force_recreate,
+                suffix=detector,
+                debug=debug,
+            )
```

Comment:
I would create class instances of the handlers, and expose functions to use here.

Reply:
could you elaborate on this?